# <img style="float: left; padding-right: 20px; width: 200px" src="https://raw.githubusercontent.com/raxlab/imt2200-data/main/media/logo.jpg">  IMT 2200 - Introducción a Ciencia de Datos
**Pontificia Universidad Católica de Chile**<br>
**Instituto de Ingeniería Matemática y Computacional**<br>
**Semestre 2025-S2**<br>
**Profesor:** Rodrigo A. Carrasco <br>

# <h1><center>Tarea 03: Exploración de datos </center></h1>

**Fecha de Entrega:** 28 de octubre de 2025, a las 23:59.

**Fecha de publicación**: 14 de octubre de 2025.

**Formato de entrega:** Notebook ejecutado y comentado (`.ipynb`) en el módulo de Tarea 03 habilitado en Canvas.

## Instrucciones

- Esta Tarea debe desarrollarse de manera totalmente *individual*, de acuerdo a lo establecido en la sección de Integridad Académica en el programa del curso.
- La Tarea debe ser desarrollada en lenguaje de programación Python y la entrega en formato Jupyter Notebook.
- El desarrollo del Notebook debe ser claro y ordenado, incluyendo anotaciones (markdown) y comentarios que permitan seguir fácilmente el código y los pasos implementados a los correctores, y siguiendo buenas prácticas de programación. La presentación y claridad del notebook y código forman parte de la evaluación de la tarea.
- Notebook **autocontenible** que:
   - Ejecute sin errores todas las celdas.
  - Contenga tanto el código como los comentarios y explicaciones necesarias.
  - Incluya visualizaciones claras y correctamente etiquetadas.
- No se aceptarán notebooks con celdas rotas o que dependan de rutas externas no indicadas en la tarea.

- Deben hacer sus consultas y comentarios sobre la Tarea a través del canal de Tareas en el Foro del curso en Canvas.

---

## 1. Objetivos

- Aplicar los conceptos iniciales de manejo de datos y análisis exploratorio vistos en clases.
- Practicar la lectura, limpieza y manipulación de datos en Python.
- Desarrollar habilidades para visualizar y describir patrones y tendencias en conjuntos de datos reales.
- Fomentar la capacidad de comunicar resultados de forma clara y fundamentada.

### 1.1 Objetivo educacional

Esta Tarea tiene como objetivo que los estudiantes desarrollen la capacidad de manejar algunas de las librerías centrales para el desarrollo de Ciencia de Datos, con foco en la lectura y exploración de datos. 

Para los ejercicios a continuación, usted deberá leer, inspeccionar, manipular y graficar conjuntos de datos en distintos formatos, de manera de responder las preguntas de cada parte de la Tarea.

### 1.2 Pregunta de ciencia de datos

En esta tarea vamos a analizar la distribución de vulneribilidad escolar para distintos establecimientos educacionales en Chile, desde enseñanza parvularia hasta enseñanza media. Específicamente, queremos entender cómo se comporta un índice de vulnerabilidad en relación a variables como ruralidad, tipo de dependencia y comuna del establecimiento.

---

## 2. Datos

En este estudio, trabajaremos con los resultados publicados anualmente sobre por la JUNAEB
sobre vulnerabilidad de establecimientos educacionales en todo el territorio nacional. Estos resultados están disponibles en la siguiente página web:

* Índices de Vulnerabilidad (JUNAEB): https://www.junaeb.cl/medicion-la-vulnerabilidad-ivm/

Adicionalmente, para visualizar las variables, utilizaremos los mapas vectoriales que disponibiliza la Biblioteca del Congreso Nacional (BCN):

* Mapoteca BCN: https://www.bcn.cl/siit/mapas_vectoriales/index_html

### 2.1 Descarga de datos y conceptos generales (1 punto)

Acceda a los links entregados y descargue el set de datos del **Índice de Vulnerabilidad Multidimensional de Establecimientos Educacionales** para el año 2025. Este debería ser un documento XLSX con múltiples pestañas, incluyendo entre estas un índice que describe las columnas presentes en la base de datos.
Descargue además los archivos correspondientes a la **División comunal: polígonos de las comunas de Chile**. Este se descarga en formato `.zip` y contiene archivos shapefile para visualizar las comunas del país.

Investigue sobre este índice y responda de manera concisa: 
* ¿En qué consiste el IVM? ¿Qué rango de valores toma y qué significan? 
* ¿Cómo se obtiene el IVM de un establecimiento educacional?



Respuesta:

## 3. Lectura y limpieza de datos (1 punto)

### 3.1 Datos de establecimientos (0.5 puntos)
El archivo descargado contiene información sobre establecimientos de cuatro tipos: Parvularia, Primer Ciclo Básica, Segundo Ciclo Básica y Media.
Cargue los datos de los establecimientos en cuatro DataFrames distintos.

Revise cada uno de sus sets de datos y haga un proceso de limpieza si lo considera necesario (manejo de valores nulos, duplicados, inválidos, transformación de tipos de datos, etc). De haberlos, identifique qué atributos son categóricos y transforme los tipos de columna de acuerdo a su decisión, mostrando todos los posibles valores de cada categoría.

Justifique **todas** sus decisiones, **incluso** si decide no realizar una limpieza o transformación.

Finalmente, responda:

* a. ¿Cuántos establecimientos tiene cada set de datos?
* b. ¿Cuánto espacio en memoria ocupan en total los 4 datasets?
* c. ¿Cuántos estudiantes fueron evaluados en todo el país para cada nivel de enseñanza para el año 2025?


In [1]:
import pandas as pd
import geopandas as gp
import matplotlib.pyplot as mp

In [2]:
# Respuestas
df_p = pd.read_excel("data/IVM_Establecimientos_2025-2.xlsx",sheet_name="Parvularia")
df_p

,ID_RBD,DV_RBD,DS_NOM_ESTABLE,DS_TIPO_DEPENDENCIA,DS_RURALIDAD,ID_REGION_ESTABLE,ID_PROVINCIA_ESTABLE,ID_COMUNA_ESTABLE,DS_COMUNA_ESTABLE,N EVALUADO,...,IVM Bajo,IVM Medio,IVM Alto,IVM Muy Alto,IVM Salud,IVM CSE Familiar,IVM Estimulación y Apoyo,IVM Familia,IVM CSE Comunal,IVM Derechos
0,2,7,PARVULARIO LAS ESPIGUITAS,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,75,...,41,18,10,6,24.494117,33.111049,34.732329,71.339647,13.929126,28.934927
1,8,6,COLEGIO INTEGRADO EDUARDO FREI MONTALVA,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,29,...,6,9,8,6,33.627244,44.572505,44.160840,70.761131,13.941963,25.186786
2,9,4,ESCUELA REPUBLICA DE ISRAEL,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,194,...,96,52,36,10,27.859852,35.035981,39.835498,70.725139,14.035447,26.995586
3,10,8,ESCUELA REPUBLICA DE FRANCIA,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,16,...,5,4,6,1,44.770271,43.530614,51.977467,71.969637,14.231790,29.929321
4,11,6,ESC. GRAL. PEDRO LAGOS MARCHANT,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,45,...,6,8,11,20,27.819437,50.101269,46.141877,67.164913,13.958703,34.009262
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
6158,42130,8,ESCUELA HOSPITALARIA PROVINCIA CORDILLERA PUEN...,Particular Subvencionado,Urbano,13,132,13201,Puente Alto,1,...,0,0,0,1,16.538717,47.820405,52.871691,88.973517,1.665128,0.000000
6159,42152,9,ESCUELA ESPECIAL DE LENGUAJE MÁGICA MELODIA,Particular Subvencionado,Urbano,4,41,4101,La Serena,16,...,6,1,4,5,24.660371,40.126775,34.230659,72.606946,11.474442,9.471729
6160,42153,7,ESCUELA ESPECIAL DE LENGUAJE IERUBA,Particular Subvencionado,Rural,4,41,4106,Vicuña,28,...,6,9,8,5,33.412175,42.794269,74.747417,69.635934,28.764205,4.436649
6161,42170,7,ESCUELA ESPECIAL DE LENGUAJE QISPI KAY,Particular Subvencionado,Urbano,4,41,4101,La Serena,12,...,4,6,1,1,40.212818,42.335762,62.300820,62.527145,11.642959,18.943457


In [3]:
df_p.value_counts()

ID_RBD  DV_RBD  DS_NOM_ESTABLE                                         DS_TIPO_DEPENDENCIA               DS_RURALIDAD  ID_REGION_ESTABLE  ID_PROVINCIA_ESTABLE  ID_COMUNA_ESTABLE  DS_COMUNA_ESTABLE  N EVALUADO  IVM Establecimiento  IVM Bajo  IVM Medio  IVM Alto  IVM Muy Alto  IVM Salud  IVM CSE Familiar  IVM Estimulación y Apoyo  IVM Familia  IVM CSE Comunal  IVM Derechos
2       7       PARVULARIO LAS ESPIGUITAS                              Servicio Local de Educación(SLE)  Urbano        15                 151                   15101              Arica              75          46.764512            41        18         10        6             24.494117  33.111049         34.732329                 71.339647    13.929126        28.934927       1
8       6       COLEGIO INTEGRADO EDUARDO FREI MONTALVA                Servicio Local de Educación(SLE)  Urbano        15                 151                   15101              Arica              29          53.177836            6         9     

In [4]:
duplicados_p= df_p.duplicated().copy()
duplicados_p #no tiene duplciados

0       False
1       False
2       False
3       False
4       False
        ...  
6158    False
6159    False
6160    False
6161    False
6162    False
Length: 6163, dtype: bool

In [5]:
df_p.isna().sum() # esta limpio

ID_RBD                      0
DV_RBD                      0
DS_NOM_ESTABLE              0
DS_TIPO_DEPENDENCIA         0
DS_RURALIDAD                0
ID_REGION_ESTABLE           0
ID_PROVINCIA_ESTABLE        0
ID_COMUNA_ESTABLE           0
DS_COMUNA_ESTABLE           0
N EVALUADO                  0
IVM Establecimiento         0
IVM Bajo                    0
IVM Medio                   0
IVM Alto                    0
IVM Muy Alto                0
IVM Salud                   0
IVM CSE Familiar            0
IVM Estimulación y Apoyo    0
IVM Familia                 0
IVM CSE Comunal             0
IVM Derechos                0
dtype: int64

In [6]:
df_pc = pd.read_excel("data/IVM_Establecimientos_2025-2.xlsx",sheet_name="1º Ciclo Básico")
df_pc

,ID_RBD,DV_RBD,DS_NOM_ESTABLE,DS_TIPO_DEPENDENCIA,DS_RURALIDAD,ID_REGION_ESTABLE,ID_PROVINCIA_ESTABLE,ID_COMUNA_ESTABLE,DS_COMUNA_ESTABLE,N EVALUADO,...,IVM Alto,IVM Muy Alto,IVM Salud,IVM Discapacidad,IVM Educación,IVM Habitabilidad,IVM Familiar,IVM Participación,IVM CSE Familia,IVM CSE Comunal
0,5,1,JOVINA NARANJO FERNANDEZ,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,106,...,26,32,47.884413,1.332290,14.028935,20.943986,37.199066,3.757660,46.502991,22.175271
1,8,6,COLEGIO INTEGRADO EDUARDO FREI MONTALVA,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,116,...,27,61,46.476438,1.704412,12.357032,21.373001,32.552305,19.197948,55.551860,22.175271
2,9,4,ESCUELA REPUBLICA DE ISRAEL,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,598,...,168,97,47.382661,3.583210,13.541001,20.168098,34.289327,1.168830,45.523334,22.175271
3,10,8,ESCUELA REPUBLICA DE FRANCIA,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,67,...,11,48,48.439375,2.950922,13.445685,21.614072,38.565676,30.706885,56.196232,22.175271
4,11,6,ESC. GRAL. PEDRO LAGOS MARCHANT,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,136,...,24,76,42.747366,5.528770,16.552890,23.111227,36.179303,20.266891,56.864454,22.175271
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
7077,42107,3,ESCUELA HOSPITALARIA UNIVERSIDAD DEL BIO-BIO,Particular Subvencionado,Urbano,16,161,16101,Chillán,3,...,0,1,33.840681,23.918486,49.439074,16.567372,59.175952,0.000000,55.770410,18.541271
7078,42130,8,ESCUELA HOSPITALARIA PROVINCIA CORDILLERA PUEN...,Particular Subvencionado,Urbano,13,132,13201,Puente Alto,14,...,2,4,37.315300,18.320649,15.184057,19.173683,40.007145,1.628601,52.403009,8.068684
7079,42154,5,ESCUELA HOSPITALARIA ENTRE VALLES,Particular Subvencionado,Urbano,4,42,4204,Salamanca,1,...,1,0,31.937012,0.000000,80.641978,12.578436,36.757006,0.000000,47.705699,33.929562
7080,42254,1,ESCUELA HOSPITALARIA DE LONCOCHE,Particular Subvencionado,Urbano,9,91,9109,Loncoche,5,...,1,1,31.937012,0.000000,17.125804,22.893818,25.163152,1.302881,52.867114,29.757018


In [7]:
df_pc.isna().sum() #no nulos

ID_RBD                  0
DV_RBD                  0
DS_NOM_ESTABLE          0
DS_TIPO_DEPENDENCIA     0
DS_RURALIDAD            0
ID_REGION_ESTABLE       0
ID_PROVINCIA_ESTABLE    0
ID_COMUNA_ESTABLE       0
DS_COMUNA_ESTABLE       0
N EVALUADO              0
IVM Establecimiento     0
IVM Bajo                0
IVM Medio               0
IVM Alto                0
IVM Muy Alto            0
IVM Salud               0
IVM Discapacidad        0
IVM Educación           0
IVM Habitabilidad       0
IVM Familiar            0
IVM Participación       0
IVM CSE Familia         0
IVM CSE Comunal         0
dtype: int64

In [8]:
duplicados_pc = df_pc.duplicated().copy()
duplicados_pc #no duplicados

0       False
1       False
2       False
3       False
4       False
        ...  
7077    False
7078    False
7079    False
7080    False
7081    False
Length: 7082, dtype: bool

In [9]:
df_sc = pd.read_excel("data/IVM_Establecimientos_2025-2.xlsx",sheet_name="2º Ciclo Básico")
df_sc

,ID_RBD,DV_RBD,DS_NOM_ESTABLE,DS_TIPO_DEPENDENCIA,DS_RURALIDAD,ID_REGION_ESTABLE,ID_PROVINCIA_ESTABLE,ID_COMUNA_ESTABLE,DS_COMUNA_ESTABLE,N EVALUADO,IVM Establecimiento,IVM Bajo,IVM Medio,IVM Alto,IVM Muy Alto
0,5,1,JOVINA NARANJO FERNANDEZ,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,60,33.190073,59,0,0,1
1,8,6,COLEGIO INTEGRADO EDUARDO FREI MONTALVA,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,81,43.372432,53,4,6,18
2,9,4,ESCUELA REPUBLICA DE ISRAEL,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,333,32.688566,328,1,0,4
3,10,8,ESCUELA REPUBLICA DE FRANCIA,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,49,41.111759,36,4,5,4
4,11,6,ESC. GRAL. PEDRO LAGOS MARCHANT,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,110,40.939232,81,11,9,9
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5657,42105,7,COLEGIO HOSPITALARIO ABRAZOSONRISAS-HOSPITAL D...,Particular Subvencionado,Urbano,13,135,13503,Curacaví,9,36.946549,9,0,0,0
5658,42107,3,ESCUELA HOSPITALARIA UNIVERSIDAD DEL BIO-BIO,Particular Subvencionado,Urbano,16,161,16101,Chillán,2,43.255081,1,0,0,1
5659,42130,8,ESCUELA HOSPITALARIA PROVINCIA CORDILLERA PUEN...,Particular Subvencionado,Urbano,13,132,13201,Puente Alto,12,39.030912,10,0,2,0
5660,42254,1,ESCUELA HOSPITALARIA DE LONCOCHE,Particular Subvencionado,Urbano,9,91,9109,Loncoche,2,31.075517,2,0,0,0


In [10]:
df_pc.isna().sum() #no nulos

ID_RBD                  0
DV_RBD                  0
DS_NOM_ESTABLE          0
DS_TIPO_DEPENDENCIA     0
DS_RURALIDAD            0
ID_REGION_ESTABLE       0
ID_PROVINCIA_ESTABLE    0
ID_COMUNA_ESTABLE       0
DS_COMUNA_ESTABLE       0
N EVALUADO              0
IVM Establecimiento     0
IVM Bajo                0
IVM Medio               0
IVM Alto                0
IVM Muy Alto            0
IVM Salud               0
IVM Discapacidad        0
IVM Educación           0
IVM Habitabilidad       0
IVM Familiar            0
IVM Participación       0
IVM CSE Familia         0
IVM CSE Comunal         0
dtype: int64

In [11]:
duplicados_sc = df_sc.duplicated().copy()
duplicados_sc #no duplicados

0       False
1       False
2       False
3       False
4       False
        ...  
5657    False
5658    False
5659    False
5660    False
5661    False
Length: 5662, dtype: bool

In [12]:
df_m = pd.read_excel("data/IVM_Establecimientos_2025-2.xlsx",sheet_name="Media")
df_m

,ID_RBD,DV_RBD,DS_NOM_ESTABLE,DS_TIPO_DEPENDENCIA,DS_RURALIDAD,ID_REGION_ESTABLE,ID_PROVINCIA_ESTABLE,ID_COMUNA_ESTABLE,DS_COMUNA_ESTABLE,N EVALUADO,IVM Establecimiento,IVM Bajo,IVM Medio,IVM Alto,IVM Muy Alto
0,1,9,LICEO POLITECNICO ARICA,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,674,27.530664,165,40,62,407
1,4,3,LICEO OCTAVIO PALMA PEREZ,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,1224,14.893692,979,65,48,132
2,5,1,JOVINA NARANJO FERNANDEZ,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,482,19.406501,270,49,39,124
3,7,8,L. POLI. BICENTENARIO DE EXCELENCIA ANTONIO VA...,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,1114,24.098877,442,77,73,522
4,8,6,COLEGIO INTEGRADO EDUARDO FREI MONTALVA,Servicio Local de Educación(SLE),Urbano,15,151,15101,Arica,404,23.462112,168,27,25,184
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2613,42090,5,ESCUELA HOSPITALARIA CLUB DE LEONES,Particular Subvencionado,Urbano,11,111,11101,Coyhaique,7,25.188487,3,0,0,4
2614,42105,7,COLEGIO HOSPITALARIO ABRAZOSONRISAS-HOSPITAL D...,Particular Subvencionado,Urbano,13,135,13503,Curacaví,26,28.028102,6,4,3,13
2615,42130,8,ESCUELA HOSPITALARIA PROVINCIA CORDILLERA PUEN...,Particular Subvencionado,Urbano,13,132,13201,Puente Alto,25,23.285271,7,4,1,13
2616,42254,1,ESCUELA HOSPITALARIA DE LONCOCHE,Particular Subvencionado,Urbano,9,91,9109,Loncoche,17,20.645259,7,2,1,7


In [13]:
df_m.isna().sum() #no nulos

ID_RBD                  0
DV_RBD                  0
DS_NOM_ESTABLE          0
DS_TIPO_DEPENDENCIA     0
DS_RURALIDAD            0
ID_REGION_ESTABLE       0
ID_PROVINCIA_ESTABLE    0
ID_COMUNA_ESTABLE       0
DS_COMUNA_ESTABLE       0
N EVALUADO              0
IVM Establecimiento     0
IVM Bajo                0
IVM Medio               0
IVM Alto                0
IVM Muy Alto            0
dtype: int64

In [14]:
duplicados_m = df_m.duplicated().copy()
duplicados_m #no duplicados

0       False
1       False
2       False
3       False
4       False
        ...  
2613    False
2614    False
2615    False
2616    False
2617    False
Length: 2618, dtype: bool

In [15]:
df_p["ID_RBD"].value_counts()

ID_RBD
2        1
8        1
9        1
10       1
11       1
        ..
42130    1
42152    1
42153    1
42170    1
42243    1
Name: count, Length: 6163, dtype: int64

In [16]:
df_pc["ID_RBD"].value_counts()

ID_RBD
5        1
8        1
9        1
10       1
11       1
        ..
42107    1
42130    1
42154    1
42254    1
42256    1
Name: count, Length: 7082, dtype: int64

In [17]:
df_sc["ID_RBD"].value_counts()

ID_RBD
5        1
8        1
9        1
10       1
11       1
        ..
42105    1
42107    1
42130    1
42254    1
42256    1
Name: count, Length: 5662, dtype: int64

In [18]:
df_m["ID_RBD"].value_counts()

ID_RBD
1        1
4        1
5        1
7        1
8        1
        ..
42090    1
42105    1
42130    1
42254    1
42256    1
Name: count, Length: 2618, dtype: int64

a)
en el df_p hay 6163 establecimientos segun la id unica
en el df_pc hay 7082 establecimientos
en el df_sc hay 5662 establecimientos
en el df_m hay 2618 establecimientos 

### 3.2 Datos geográficos (0.5 puntos)

Cargue en un GeoDataFrame de GeoPandas la información de los polígonos de cada comuna del país.

Como realizamos en la Tarea pasada, puede transformar el Coordinate Reference System (CRS) a EPSG 4326, para tener la información de los puntos de los polígonos con latitud y longitud.

In [19]:
# Respuesta
gdf = gp.read_file("data/comunas.zip")
gdf

,objectid,shape_leng,dis_elec,cir_sena,cod_comuna,codregion,st_area_sh,st_length_,Region,Comuna,Provincia,geometry
0,48,170038.624165,16,8,6204,6,9.685774e+08,206184.271675,Región del Libertador Bernardo O'Higgins,Marchigüe,Cardenal Caro,"POLYGON ((-7992818.789 -4056669.319, -7992731...."
1,29,125730.104795,15,8,6102,6,4.157446e+08,151911.576827,Región del Libertador Bernardo O'Higgins,Codegua,Cachapoal,"POLYGON ((-7831651.813 -4022443.109, -7831596...."
2,30,63026.084422,15,8,6103,6,1.448565e+08,76355.326122,Región del Libertador Bernardo O'Higgins,Coinco,Cachapoal,"POLYGON ((-7892616.357 -4060673.364, -7892621...."
3,31,89840.903562,15,8,6104,6,3.256572e+08,108874.623150,Región del Libertador Bernardo O'Higgins,Coltauco,Cachapoal,"POLYGON ((-7906457.717 -4051723.734, -7906266...."
4,78,122626.493264,23,11,9121,9,6.990727e+08,156680.410681,Región de La Araucanía,Cholchol,Cautín,"POLYGON ((-8121756.186 -4645860.295, -8121644...."
...,...,...,...,...,...,...,...,...,...,...,...,...
341,255,130240.555805,19,10,16305,16,8.751006e+08,160459.137929,Región de Ñuble,San Nicolás,Punilla,"POLYGON ((-8041950.854 -4349201.562, -8041735...."
342,253,311061.932415,19,10,16304,16,2.393007e+09,388109.613857,Región de Ñuble,San Fabián,Punilla,"POLYGON ((-7931328.549 -4363544.179, -7931320...."
343,245,159257.906431,19,10,16303,16,7.638173e+08,195593.933727,Región de Ñuble,Ñiquén,Punilla,"POLYGON ((-8026032.186 -4319505.012, -8025933...."
344,243,157467.049667,19,10,16104,16,1.044405e+09,196962.664313,Región de Ñuble,El Carmen,Ñuble,"POLYGON ((-7986315.104 -4415272.935, -7986309...."


In [20]:
gdf_crs= gdf.to_crs(epsg=4326) # transformamos a epsg
gdf_crs

,objectid,shape_leng,dis_elec,cir_sena,cod_comuna,codregion,st_area_sh,st_length_,Region,Comuna,Provincia,geometry
0,48,170038.624165,16,8,6204,6,9.685774e+08,206184.271675,Región del Libertador Bernardo O'Higgins,Marchigüe,Cardenal Caro,"POLYGON ((-71.80071 -34.20728, -71.79992 -34.2..."
1,29,125730.104795,15,8,6102,6,4.157446e+08,151911.576827,Región del Libertador Bernardo O'Higgins,Codegua,Cachapoal,"POLYGON ((-70.35293 -33.95263, -70.35243 -33.9..."
2,30,63026.084422,15,8,6103,6,1.448565e+08,76355.326122,Región del Libertador Bernardo O'Higgins,Coinco,Cachapoal,"POLYGON ((-70.90058 -34.23703, -70.90063 -34.2..."
3,31,89840.903562,15,8,6104,6,3.256572e+08,108874.623150,Región del Libertador Bernardo O'Higgins,Coltauco,Cachapoal,"POLYGON ((-71.02492 -34.17053, -71.0232 -34.17..."
4,78,122626.493264,23,11,9121,9,6.990727e+08,156680.410681,Región de La Araucanía,Cholchol,Cautín,"POLYGON ((-72.95898 -38.46877, -72.95797 -38.4..."
...,...,...,...,...,...,...,...,...,...,...,...,...
341,255,130240.555805,19,10,16305,16,8.751006e+08,160459.137929,Región de Ñuble,San Nicolás,Punilla,"POLYGON ((-72.24207 -36.35227, -72.24014 -36.3..."
342,253,311061.932415,19,10,16304,16,2.393007e+09,388109.613857,Región de Ñuble,San Fabián,Punilla,"POLYGON ((-71.24834 -36.45596, -71.24827 -36.4..."
343,245,159257.906431,19,10,16303,16,7.638173e+08,195593.933727,Región de Ñuble,Ñiquén,Punilla,"POLYGON ((-72.09907 -36.13712, -72.09819 -36.1..."
344,243,157467.049667,19,10,16104,16,1.044405e+09,196962.664313,Región de Ñuble,El Carmen,Ñuble,"POLYGON ((-71.74229 -36.82882, -71.74224 -36.8..."


## 4. Análisis descriptivo (1.5 puntos)

En esta sección haremos una exploración preliminar de los 4 sets de datos cargados. Para cada una de las siguientes preguntas, sustente su respuesta con al menos un gráfico y su análisis respectivo.

**Recomendación:** puede utilizar el método `subplots()` de `matplotlib.pyplot` para facilitar la visualización de múltiples gráficos a la vez.

### 4.1 Ruralidad (0.8 pts)
* a. ¿Cuál es la cantidad de establecimientos rurales vs urbanos según nivel de enseñanza?
* b. ¿Cómo se distribuye el IVM según nivel de enseñanza y ruralidad?


In [21]:
df_p["DS_RURALIDAD"].value_counts()

DS_RURALIDAD
Urbano    4903
Rural     1260
Name: count, dtype: int64

In [22]:
df_pc["DS_RURALIDAD"].value_counts()

DS_RURALIDAD
Urbano    4128
Rural     2954
Name: count, dtype: int64

In [23]:
df_sc["DS_RURALIDAD"].value_counts()

DS_RURALIDAD
Urbano    4170
Rural     1492
Name: count, dtype: int64

In [24]:
df_m["DS_RURALIDAD"].value_counts()

DS_RURALIDAD
Urbano    2449
Rural      169
Name: count, dtype: int64

In [25]:
df_p_u= df_p[df_p["DS_RURALIDAD"] == "Urbano"]
df_p_r= df_p[df_p["DS_RURALIDAD"] == "Rural"]
df_pc_u= df_pc[df_pc["DS_RURALIDAD"] == "Urbano"]
df_pc_r= df_pc[df_pc["DS_RURALIDAD"] == "Rural"]
df_sc_u= df_sc[df_sc["DS_RURALIDAD"] == "Urbano"]
df_sc_r= df_sc[df_sc["DS_RURALIDAD"] == "Rural"]
df_m_u= df_m[df_m["DS_RURALIDAD"] == "Urbano"]
df_m_r= df_m[df_m["DS_RURALIDAD"] == "Rural"]

In [26]:
p_b_r = df_p_r["IVM Bajo"].sum()
p_m_r =df_p_r["IVM Medio"].sum()
p_a_r =df_p_r["IVM Alto"].sum()
print(p_b_r,p_m_r,p_a_r)

3752 5146 7048


podemo ver que es mas equilibrado en parbulo rural

In [27]:
p_b_u = df_p_u["IVM Bajo"].sum()
p_m_u =df_p_u["IVM Medio"].sum()
p_a_u =df_p_u["IVM Alto"].sum()
print(p_b_u,p_m_u,p_a_u)

60125 58730 56828


aqui se puede ver que en parbulo urbano el hay menos segun mas alto el ivm

In [28]:
p_bp_r = df_pc_r["IVM Bajo"].sum()
p_mp_r =df_pc_r["IVM Medio"].sum()
p_ap_r =df_pc_r["IVM Alto"].sum()
print(p_bp_r,p_mp_r,p_ap_r)

25870 27065 27823


en primer ciclo basico se puede ver relativamente equilibrado pero mayor entre ivm medio y alto con menos ivm bajo en los establecimientos rurales

In [29]:
p_bp_u = df_pc_u["IVM Bajo"].sum()
p_mp_u =df_pc_u["IVM Medio"].sum()
p_ap_u =df_pc_u["IVM Alto"].sum()
print(p_bp_u,p_mp_u,p_ap_u)

194850 193655 192897


mientras que en urbano diria que esta relativamente equilibrado

In [30]:
p_bs_r = df_sc_r["IVM Bajo"].sum()
p_ms_r =df_sc_r["IVM Medio"].sum()
p_as_r =df_sc_r["IVM Alto"].sum()
print(p_bs_r,p_ms_r,p_as_r)

41129 2459 2132


en segundo ciclo basico vemos que hay mucho mas ivm bajo que medio y alto en los establecimientos rurales

In [31]:
p_bs_u = df_sc_u["IVM Bajo"].sum()
p_ms_u =df_sc_u["IVM Medio"].sum()
p_as_u =df_sc_u["IVM Alto"].sum()
print(p_bs_u,p_ms_u,p_as_u)

344395 28261 25892


mientras que en los urbanos la diferencia no es tan abrumadora pero sigue siendo alta y continua el patron de a mas alto ivm menos hay 

In [32]:
p_bm_r = df_m_r["IVM Bajo"].sum()
p_mm_r =df_m_r["IVM Medio"].sum()
p_am_r =df_m_r["IVM Alto"].sum()
print(p_bm_r,p_mm_r,p_am_r)

16448 3205 2907


en la media se sigue el mismo patron anterior de una diferencia substancial entre bajo y los demas pero sigue bajando segun mas alto el ivm en los establecimientos rurales

In [33]:
p_bm_u = df_m_u["IVM Bajo"].sum()
p_mm_u =df_m_u["IVM Medio"].sum()
p_am_u =df_m_u["IVM Alto"].sum()
print(p_bm_u,p_mm_u,p_am_u)

424191 82329 73094


aqui es mas extrano ya que el mayor se encuentra en el ivm medio, sigue alto y en ultimo bajo en los establecimientos urbanos

Respuesta y comentarios:

podemos ver que segun el nivel entre basica y media la cantidad de establecimientos rurales se reduce, mientras que entre parbulo y primer ciclo basico sube

### 4.2 Tipo de establecimiento (0.7 pts)

* ¿Cómo es la distribución del IVM promedio de los establecimientos según tipo de dependencia y nivel de enseñanza? 

In [34]:
# respuestas
df_p["DS_TIPO_DEPENDENCIA"].value_counts()

DS_TIPO_DEPENDENCIA
Particular Subvencionado            3428
Municipal DAEM                      1510
Servicio Local de Educación(SLE)     854
Corporación Municipal                371
Name: count, dtype: int64

en parbulo se puede ver que particular subvencionado es el que mas se muestra, seguido de municipal daem, sle y corporacon municipal

In [35]:
df_p_c = df_p[df_p["DS_TIPO_DEPENDENCIA"] == "Corporación Municipal"]
df_p_p = df_p[df_p["DS_TIPO_DEPENDENCIA"] == "Particular Subvencionado"]
df_p_m = df_p[df_p["DS_TIPO_DEPENDENCIA"] == "Municipal DAEM"]
df_p_s = df_p[df_p["DS_TIPO_DEPENDENCIA"] == "Servicio Local de Educación(SLE)"]
df_pc_c = df_pc[df_pc["DS_TIPO_DEPENDENCIA"] == "Corporación Municipal"]
df_pc_p = df_pc[df_pc["DS_TIPO_DEPENDENCIA"] == "Particular Subvencionado"]
df_pc_m = df_pc[df_pc["DS_TIPO_DEPENDENCIA"] == "Municipal DAEM"]
df_pc_s = df_pc[df_pc["DS_TIPO_DEPENDENCIA"] == "Servicio Local de Educación(SLE)"]
df_pc_a = df_pc[df_pc["DS_TIPO_DEPENDENCIA"] == "Administración Delegada (C. P. 3166)"]
df_sc_c = df_sc[df_sc["DS_TIPO_DEPENDENCIA"] == "Corporación Municipal"]
df_sc_p = df_sc[df_sc["DS_TIPO_DEPENDENCIA"] == "Particular Subvencionado"]
df_sc_m = df_sc[df_sc["DS_TIPO_DEPENDENCIA"] == "Municipal DAEM"]
df_sc_s = df_sc[df_sc["DS_TIPO_DEPENDENCIA"] == "Servicio Local de Educación(SLE)"]
df_sc_a = df_sc[df_sc["DS_TIPO_DEPENDENCIA"] == "Administración Delegada (C. P. 3166)"]
df_m_c = df_m[df_m["DS_TIPO_DEPENDENCIA"] == "Corporación Municipal"]
df_m_p = df_m[df_m["DS_TIPO_DEPENDENCIA"] == "Particular Subvencionado"]
df_m_m = df_m[df_m["DS_TIPO_DEPENDENCIA"] == "Municipal DAEM"]
df_m_s = df_m[df_m["DS_TIPO_DEPENDENCIA"] == "Servicio Local de Educación(SLE)"]
df_m_a = df_m[df_m["DS_TIPO_DEPENDENCIA"] == "Administración Delegada (C. P. 3166)"]


In [36]:
# Corporación Municipal
p_c_b = df_p_c["IVM Bajo"]
p_c_m = df_p_c["IVM Medio"]
p_c_a = df_p_c["IVM Alto"]
# Particular Subvencionado
p_p_b = df_p_p["IVM Bajo"]
p_p_m = df_p_p["IVM Medio"]
p_p_a = df_p_p["IVM Alto"]
# Municipal DAEM
p_m_b = df_p_m["IVM Bajo"]
p_m_m = df_p_m["IVM Medio"]
p_m_a = df_p_m["IVM Alto"]
# Servicio Local de Educación(SLE)
p_s_b = df_p_s["IVM Bajo"]
p_s_m = df_p_s["IVM Medio"]
p_s_a = df_p_s["IVM Alto"]



para primer ciclo

In [37]:
# Corporación Municipal
pc_c_b = df_pc_c["IVM Bajo"]
pc_c_m = df_pc_c["IVM Medio"]
pc_c_a = df_pc_c["IVM Alto"]
# Particular Subvencionado
pc_p_b = df_pc_p["IVM Bajo"]
pc_p_m = df_pc_p["IVM Medio"]
pc_p_a = df_pc_p["IVM Alto"]
# Municipal DAEM
pc_m_b = df_pc_m["IVM Bajo"]
pc_m_m = df_pc_m["IVM Medio"]
pc_m_a = df_pc_m["IVM Alto"]
# Servicio Local de Educación(SLE)
pc_s_b = df_pc_s["IVM Bajo"]
pc_s_m = df_pc_s["IVM Medio"]
pc_s_a = df_pc_s["IVM Alto"]
# Administración Delegada
pc_a_b = df_pc_a["IVM Bajo"]
pc_a_m = df_pc_a["IVM Medio"]
pc_a_a = df_pc_a["IVM Alto"]


para segundo ciclo

In [38]:
# Corporación Municipal
sc_c_b = df_sc_c["IVM Bajo"]
sc_c_m = df_sc_c["IVM Medio"]
sc_c_a = df_sc_c["IVM Alto"]
# Particular Subvencionado
sc_p_b = df_sc_p["IVM Bajo"]
sc_p_m = df_sc_p["IVM Medio"]
sc_p_a = df_sc_p["IVM Alto"]
# Municipal DAEM
sc_m_b = df_sc_m["IVM Bajo"]
sc_m_m = df_sc_m["IVM Medio"]
sc_m_a = df_sc_m["IVM Alto"]
# Servicio Local de Educación(SLE)
sc_s_b = df_sc_s["IVM Bajo"]
sc_s_m = df_sc_s["IVM Medio"]
sc_s_a = df_sc_s["IVM Alto"]
# Administración Delegada
sc_a_b = df_sc_a["IVM Bajo"]
sc_a_m = df_sc_a["IVM Medio"]
sc_a_a = df_sc_a["IVM Alto"]


para media

In [39]:
# Corporación Municipal
m_c_b = df_m_c["IVM Bajo"]
m_c_m = df_m_c["IVM Medio"]
m_c_a = df_m_c["IVM Alto"]
# Particular Subvencionado
m_p_b = df_m_p["IVM Bajo"]
m_p_m = df_m_p["IVM Medio"]
m_p_a = df_m_p["IVM Alto"]
# Municipal DAEM
m_m_b = df_m_m["IVM Bajo"]
m_m_m = df_m_m["IVM Medio"]
m_m_a = df_m_m["IVM Alto"]
# Servicio Local de Educación(SLE)
m_s_b = df_m_s["IVM Bajo"]
m_s_m = df_m_s["IVM Medio"]
m_s_a = df_m_s["IVM Alto"]
# Administración Delegada
m_a_b = df_m_a["IVM Bajo"]
m_a_m = df_m_a["IVM Medio"]
m_a_a = df_m_a["IVM Alto"]


corporación Municipal

In [40]:
p_c_b.sum()

np.int64(4678)

In [41]:
p_c_m.sum()

np.int64(4827)

In [42]:
p_c_a.sum()

np.int64(4788)

en parbulo con dependencia de corporacion municipal es equilibrado entre todos los ivm

particular Subvencionado

In [43]:
p_p_b.sum()

np.int64(44626)

In [44]:
p_p_m.sum()

np.int64(41186)

In [45]:
p_p_a.sum()

np.int64(38790)

aqui podemos ver que para parbulo en particular subvencionado la cantidad baja segund mas alto el ivm 

municipal DAEM

In [46]:
p_m_b.sum()

np.int64(8124)

In [47]:
p_m_m.sum()

np.int64(10082)

In [48]:
p_m_a.sum()

np.int64(12133)

en parbulario en municipal daem podemosver que al mas alto ivm mayor cantidad hay 

servicio Local de Educación(SLE)

In [49]:
p_s_b.sum()

np.int64(6449)

In [50]:
p_s_m.sum()

np.int64(7781)

In [51]:
p_s_a.sum()

np.int64(8165)

para parbulo en sle sube segun mas alto el ivm

primer ciclo

corporación Municipal

In [52]:
pc_c_b.sum()

np.int64(13491)

In [53]:
pc_c_m.sum()

np.int64(15082)

In [54]:
pc_c_a.sum()

np.int64(16754)

en primer ciclo corporacion municipal podemos ver que tambien sube segun lo alto del ivm

particular Subvencionado

In [55]:
pc_p_b.sum()

np.int64(147257)

In [56]:
pc_p_m.sum()

np.int64(137861)

In [57]:
pc_p_a.sum()

np.int64(128510)

en primer ciclo particular subvencionado baja segun mas altura del ivm

municipal DAEM

In [58]:
pc_m_b.sum()

np.int64(37445)

In [59]:
pc_m_m.sum()

np.int64(41317)

In [60]:
pc_m_a.sum()

np.int64(44902)

en primer ciclo municipalidad daem podemos ver que sube segun mas alto el ivm

servicio Local de Educación(SLE)

In [61]:
pc_s_b.sum()

np.int64(22484)

In [62]:
pc_s_m.sum()

np.int64(26433)

In [63]:
pc_s_a.sum()

np.int64(30515)

en primer ciclo sle podemos ver que sigue el patron de subir segun mas alto el ivm

administración Delegada

In [64]:
pc_a_b.sum()

np.int64(43)

In [65]:
pc_a_m.sum()

np.int64(27)

In [66]:
pc_a_a.sum()

np.int64(39)

en administracion delegada podemos ver que no sigue el ciclo siendo alto-bajo-medio pero son pocas muestras en este caso

segundo ciclo

corporación Municipal

In [67]:
sc_c_b.sum()

np.int64(30272)

In [68]:
sc_c_m.sum()

np.int64(3023)

In [69]:
sc_c_a.sum()

np.int64(2857)

segundo ciclo corporacion municipal la diferencia entre ivm bajo y los demas es sumamente alta

particular Subvencionado

In [70]:
sc_p_b.sum()

np.int64(221267)

In [71]:
sc_p_m.sum()

np.int64(17027)

In [72]:
sc_m_a.sum()

np.int64(5296)

lo mismo denuevo solo que  muestra general es mas altaen particular subvencionado

Municipal DAEM

In [73]:
sc_m_b.sum()

np.int64(79218)

In [74]:
sc_m_m.sum()

np.int64(6102)

In [75]:
sc_m_a.sum()

np.int64(5296)

lo mismo, bajo sumamente alto en comparacion a los demas

servicio Local de Educación(SLE)

In [76]:
sc_s_b.sum()

np.int64(54576)

In [77]:
sc_s_m.sum()

np.int64(4550)

In [78]:
sc_s_a.sum()

np.int64(4400)

devuelta al mismo patron 

Administración Delegada

In [79]:
sc_a_b.sum()

np.int64(191)

In [80]:
sc_a_m.sum()

np.int64(18)

In [81]:
sc_a_a.sum()

np.int64(17)

pese a que sea una muestra pequena aun sigue el patron 

media

Corporación Municipal

In [82]:
m_c_b.sum()

np.int64(27512)

In [83]:
m_c_m.sum()

np.int64(6058)

In [84]:
m_c_a.sum()

np.int64(5567)

muy similar a segundo ciclo ivm bajo es muyo mayor a los demas

Particular Subvencionado

In [85]:
m_p_b.sum()

np.int64(266537)

In [86]:
m_p_m.sum()

np.int64(49811)

In [87]:
m_p_a.sum()

np.int64(43578)

al igual que en segundo ciclo sigue siendo bajo mucho mas presente que los demas pero como es particular subvencionado la muestra es mucho mas grande 

Municipal DAEM

In [88]:
m_m_b.sum()

np.int64(82711)

In [89]:
m_m_m.sum()

np.int64(15520)

In [90]:
m_m_a.sum()

np.int64(13967)

mismo patron

Servicio Local de Educación(SLE)

In [91]:
m_s_b.sum()

np.int64(48821)

In [92]:
m_s_m.sum()

np.int64(9995)

In [93]:
m_s_a.sum()

np.int64(8936)

mismo patron otra vez

Administración Delegada

In [94]:
m_a_b.sum()

np.int64(15058)

In [95]:
m_a_m.sum()

np.int64(4150)

In [96]:
m_a_a.sum()

np.int64(3953)

el primer establecimiento de administracion delegada con una muestra mas grande que sigue el patron 

Respuesta y comentarios:

In [97]:
# Corporación Municipal
m_c_b = df_m_c["IVM Bajo"]
m_c_m = df_m_c["IVM Medio"]
m_c_a = df_m_c["IVM Alto"]
# Particular Subvencionado
m_p_b = df_m_p["IVM Bajo"]
m_p_m = df_m_p["IVM Medio"]
m_p_a = df_m_p["IVM Alto"]
# Municipal DAEM
m_m_b = df_m_m["IVM Bajo"]
m_m_m = df_m_m["IVM Medio"]
m_m_a = df_m_m["IVM Alto"]
# Servicio Local de Educación(SLE)
m_s_b = df_m_s["IVM Bajo"]
m_s_m = df_m_s["IVM Medio"]
m_s_a = df_m_s["IVM Alto"]
# Administración Delegada
m_a_b = df_m_a["IVM Bajo"]
m_a_m = df_m_a["IVM Medio"]
m_a_a = df_m_a["IVM Alto"]

en parbulo muestra varios casos, equilibrado en corporacion municipal, inversamente proporcional en particular subvencionado, proporcional en municipal daem y sle
en primer ciclo esto cambia ya que en todos es proporcional al nivel de ivm menos en particular subvencionado que es inversamente proporcional y en administracion delegada que no sigue un patron pero son muy pocas muestras
en segundo ciclo y media se sigue el mismo patron que en todos los tipos de depenedencia es inversamente proporcional y aparte de esto la cantidad en ivm bajo son mucho mayores a los demas niveles de ivm 

### 4.3 Pregunta Bono: Dimensiones de vulnerabilidad (1 punto)

Para los niveles de enseñanza Parvularia y Básica Primer ciclo (1-4º), tenemos información sobre distintas dimensiones de que se utilizan para calcular el IVM, como condición socioeconómica familiar y comunal, salud, etc. 

* Para todo Chile, en promedio, ¿cuál es la dimensión con mayor puntaje de IVM para cada uno de estos dos niveles de enseñanza?

In [98]:
df_p[["IVM Derechos","IVM Estimulación y Apoyo","IVM Familia","IVM Salud","IVM CSE Comunal","IVM CSE Familiar"]].sum()

IVM Derechos                 87584.181279
IVM Estimulación y Apoyo    265597.964625
IVM Familia                 435521.245002
IVM Salud                   172010.555957
IVM CSE Comunal             111720.925867
IVM CSE Familiar            266205.915500
dtype: float64

en parvulo es el ivm de familia

In [99]:
df_pc[["IVM Discapacidad","IVM Habitabilidad","IVM Familiar","IVM Salud","IVM CSE Comunal","IVM CSE Familia","IVM Participación"]].sum()

IVM Discapacidad      41959.193432
IVM Habitabilidad    146889.621326
IVM Familiar         228018.219946
IVM Salud            305250.841592
IVM CSE Comunal      160963.283033
IVM CSE Familia      385195.373972
IVM Participación     37724.394898
dtype: float64

y en primer ciclo basico es el ivm cse familia

Respuesta y comentarios:

## 5. Agrupación de bases de datos (2 puntos)

El objetivo de esta sección será agrupar los establecimientos según la comuna en la que se encuentran para obtener un resumen general sobre la vulnerabilidad escolar en cada una de ellas. Al finalizar esta sección, tendremos un nuevo DataFrame `agg_comunal` que contenga, al menos, la siguiente información:

* Comuna (nombre y ID)
* Región (ID)
* Cantidad de establecimientos rurales de la comuna
* Cantidad de establecimientos urbanos de la comuna
* Cantidad total de establecimientos de la comuna
* Porcentaje de establecimientos rurales
* IVM Promedio Parvularia de la comuna
* IVM Promedio Básica Primer Ciclo de la comuna
* IVM Promedio Básica Segundo Ciclo de la comuna
* IVM Promedio Media de la comuna
* Objeto geometry de GeoPandas correspondiente al polígono de la comuna respectiva

Armaremos este DataFrame paso a paso.

### 5.1 Cálculo de IVM promedio por comuna (0.8 pts)

Primero, para cada uno sus 4 DataFrames, calcule el promedio de la columna `IVM Establecimiento` según comuna. Almacene este valor en una nueva columna llamada `IVM Promedio {NIVEL DE ENSEÑANZA}` (por ejemplo, para enseñanza parvularia, cree la columna `IVM Promedio Parvularia`).

Junte los 4 resultados que obtuvo en un solo DataFrame `ivm_comunal` que contenga:
* Comuna (nombre y ID)
* Región (ID)
* IVM Promedio Parvularia de la comuna
* IVM Promedio Básica Primer Ciclo de la comuna
* IVM Promedio Básica Segundo Ciclo de la comuna
* IVM Promedio Media de la comuna

Ponga siempre atención al tamaño de sus DataFrames: como estamos agrupando comunalmente, estos siempre debiesen tener una cantidad de filas igual a las comunas de Chile.

In [100]:
ivm_p = df_p[["IVM Establecimiento"]].mean()
ivm_p = ivm_p.mean()
ivm_pc = df_pc[["IVM Establecimiento"]].mean()
ivm_pc = ivm_pc.mean()
ivm_sc = df_sc[["IVM Establecimiento"]].mean()
ivm_sc= ivm_sc.mean()
ivm_m = df_m[["IVM Establecimiento"]].mean()
ivm_m = ivm_m.mean()

In [101]:
df_p_com = df_p[["ID_REGION_ESTABLE","DS_COMUNA_ESTABLE","ID_COMUNA_ESTABLE"]].drop_duplicates(subset="DS_COMUNA_ESTABLE")
df_p_com["IVM Promedio Parvulo"] = ivm_p
df_p_com

,ID_REGION_ESTABLE,DS_COMUNA_ESTABLE,ID_COMUNA_ESTABLE,IVM Promedio Parvulo
0,15,Arica,15101,55.460412
33,15,Putre,15201,55.460412
34,1,Iquique,1101,55.460412
49,1,Huara,1404,55.460412
52,1,Camiña,1402,55.460412
...,...,...,...,...
3373,13,María Pinto,13504,55.460412
3385,13,San Pedro,13505,55.460412
3393,13,Alhué,13502,55.460412
3572,1,Alto Hospicio,1107,55.460412


In [102]:
df_pc_com = df_pc[["ID_REGION_ESTABLE","DS_COMUNA_ESTABLE","ID_COMUNA_ESTABLE"]].drop_duplicates(subset="DS_COMUNA_ESTABLE")
df_pc_com["IVM Promedio Primer Ciclo"] = ivm_pc

In [103]:
df_sc_com =df_sc[["ID_REGION_ESTABLE","DS_COMUNA_ESTABLE","ID_COMUNA_ESTABLE"]].drop_duplicates(subset="DS_COMUNA_ESTABLE")
df_sc_com["IVM Promedio Segundo Ciclo"] = ivm_sc
df_sc_com

,ID_REGION_ESTABLE,DS_COMUNA_ESTABLE,ID_COMUNA_ESTABLE,IVM Promedio Segundo Ciclo
0,15,Arica,15101,38.432973
32,15,Camarones,15102,38.432973
34,15,Putre,15201,38.432973
35,15,General Lagos,15202,38.432973
36,1,Iquique,1101,38.432973
...,...,...,...,...
3848,13,San Pedro,13505,38.432973
3858,13,Alhué,13502,38.432973
3870,1,Alto Hospicio,1107,38.432973
4352,5,Isla de Pascua,5201,38.432973


In [104]:
df_m_com = df_m[["ID_REGION_ESTABLE","DS_COMUNA_ESTABLE","ID_COMUNA_ESTABLE"]].drop_duplicates(subset="DS_COMUNA_ESTABLE")
df_m_com["IVM Promedio Media"] = ivm_m
df_m_com

,ID_REGION_ESTABLE,DS_COMUNA_ESTABLE,ID_COMUNA_ESTABLE,IVM Promedio Media
0,15,Arica,15101,20.252681
14,15,Camarones,15102,20.252681
15,15,Putre,15201,20.252681
16,1,Iquique,1101,20.252681
33,1,Huara,1404,20.252681
...,...,...,...,...
2159,11,Cochrane,11301,20.252681
2503,16,Treguaco,16207,20.252681
2546,4,Paiguano,4105,20.252681
2550,6,Chépica,6302,20.252681


In [105]:
ivm_comunal = pd.merge(df_p_com,df_pc_com,on = ["ID_REGION_ESTABLE","DS_COMUNA_ESTABLE","ID_COMUNA_ESTABLE"],how = "outer")
ivm_comunal = ivm_comunal.merge(df_sc_com,on = ["ID_REGION_ESTABLE","DS_COMUNA_ESTABLE","ID_COMUNA_ESTABLE"],how = "outer")
ivm_comunal = ivm_comunal.merge(df_m_com,on = ["ID_REGION_ESTABLE","DS_COMUNA_ESTABLE","ID_COMUNA_ESTABLE"],how = "outer")
ivm_comunal 

,ID_REGION_ESTABLE,DS_COMUNA_ESTABLE,ID_COMUNA_ESTABLE,IVM Promedio Parvulo,IVM Promedio Primer Ciclo,IVM Promedio Segundo Ciclo,IVM Promedio Media
0,1,Alto Hospicio,1107,55.460412,43.522262,38.432973,20.252681
1,1,Camiña,1402,55.460412,43.522262,38.432973,20.252681
2,1,Colchane,1403,55.460412,43.522262,38.432973,20.252681
3,1,Huara,1404,55.460412,43.522262,38.432973,20.252681
4,1,Iquique,1101,55.460412,43.522262,38.432973,20.252681
...,...,...,...,...,...,...,...
340,16,San Ignacio,16108,55.460412,43.522262,38.432973,20.252681
341,16,San Nicolás,16305,55.460412,43.522262,38.432973,20.252681
342,16,Treguaco,16207,55.460412,43.522262,38.432973,20.252681
343,16,Yungay,16109,55.460412,43.522262,38.432973,20.252681


In [106]:
ivm_comunal.isna().sum()

ID_REGION_ESTABLE              0
DS_COMUNA_ESTABLE              0
ID_COMUNA_ESTABLE              0
IVM Promedio Parvulo           7
IVM Promedio Primer Ciclo      0
IVM Promedio Segundo Ciclo     1
IVM Promedio Media            10
dtype: int64

junte los df de forma en la que se junten por columnas, al ver mas el df puedo ver que hay nulos en ciertas ensenanzas y eso se debe a que la cantidad de regiones en los dfs de cada ensenanza es distinto ya que hay ciertos lugares que no tienen ciertos niveles de educacion 

### 5.2 Establecimientos por comuna y ruralidad (0.8 pts)

Ahora contaremos la cantidad de establecimientos por comuna según ruralidad. Como buscamos la cantidad total de establecimientos, consideramos establecimientos de todos los niveles educacionales vistos. Para esto, combine los 4 DataFrames que cargó y genere un nuevo DataFrame `estab_comunal` que contenga:

* Comuna (ID)
* Cantidad de establecimientos rurales
* Cantidad de establecimientos urbanos
* Cantidad total de establecimientos
* Porcentaje de establecimientos rurales (Establecimientos rurales / Total establecimientos * 100)

Tenga en cuenta de que pueden aparecer establecimientos duplicados al combinar estos 4 DataFrames (por ejemplo, establecimientos que imparten múltiples niveles de enseñanza, desde 1º básico hasta 4º medio). Preocúpese de manejar estos casos antes de agrupar sus datos según ruralidad.

In [107]:
df_p_r.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1260 entries, 19 to 6160
Data columns (total 21 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   ID_RBD                    1260 non-null   int64  
 1   DV_RBD                    1260 non-null   int64  
 2   DS_NOM_ESTABLE            1260 non-null   object 
 3   DS_TIPO_DEPENDENCIA       1260 non-null   object 
 4   DS_RURALIDAD              1260 non-null   object 
 5   ID_REGION_ESTABLE         1260 non-null   int64  
 6   ID_PROVINCIA_ESTABLE      1260 non-null   int64  
 7   ID_COMUNA_ESTABLE         1260 non-null   int64  
 8   DS_COMUNA_ESTABLE         1260 non-null   object 
 9   N EVALUADO                1260 non-null   int64  
 10  IVM Establecimiento       1260 non-null   float64
 11  IVM Bajo                  1260 non-null   int64  
 12  IVM Medio                 1260 non-null   int64  
 13  IVM Alto                  1260 non-null   int64  
 14  IVM Muy Alto

In [108]:
cantidad_p_r = len(df_p_r["DS_NOM_ESTABLE"])
cantidad_p_u = len(df_p_u["DS_NOM_ESTABLE"])
cantidad_pc_r = len(df_pc_r["DS_NOM_ESTABLE"])
cantidad_pc_u = len(df_pc_u["DS_NOM_ESTABLE"])
cantidad_sc_r = len(df_sc_r["DS_NOM_ESTABLE"])
cantidad_sc_u = len(df_sc_u["DS_NOM_ESTABLE"])
cantidad_m_r = len(df_m_r["DS_NOM_ESTABLE"])
cantidad_m_u = len(df_m_u["DS_NOM_ESTABLE"])

In [109]:
df_p_r_c = df_p_r.copy()
df_p_u_c = df_p_u.copy()
df_pc_r_c = df_pc_r.copy()
df_pc_u_c = df_pc_u.copy()
df_sc_r_c = df_sc_r.copy()
df_sc_u_c = df_sc_u.copy()
df_m_r_c = df_m_r.copy()
df_m_u_c = df_m_u.copy()

In [110]:
df_p_r_c["cantidad establecimientos rurales"] = cantidad_p_r
df_p_u_c["cantidad establecimientos urbanos"] = cantidad_p_u
df_pc_r_c["cantidad establecimientos rurales"] = cantidad_pc_r
df_pc_u_c["cantidad establecimientos urbanos"] = cantidad_pc_u
df_sc_r_c["cantidad establecimientos rurales"] = cantidad_sc_r
df_sc_u_c["cantidad establecimientos urbanos"] = cantidad_sc_u
df_m_r_c["cantidad establecimientos rurales"] = cantidad_m_r
df_m_u_c["cantidad establecimientos urbanos"] = cantidad_m_u


In [111]:
df_p_r

,ID_RBD,DV_RBD,DS_NOM_ESTABLE,DS_TIPO_DEPENDENCIA,DS_RURALIDAD,ID_REGION_ESTABLE,ID_PROVINCIA_ESTABLE,ID_COMUNA_ESTABLE,DS_COMUNA_ESTABLE,N EVALUADO,...,IVM Bajo,IVM Medio,IVM Alto,IVM Muy Alto,IVM Salud,IVM CSE Familiar,IVM Estimulación y Apoyo,IVM Familia,IVM CSE Comunal,IVM Derechos
19,31,0,ESCUELA DARIO SALAS DIAZ,Servicio Local de Educación(SLE),Rural,15,151,15101,Arica,67,...,4,14,23,26,38.446642,49.815197,52.852825,74.820768,14.813061,66.318423
20,32,9,LICEO AGRICOLA JOSE ABELARDO NUNEZ M.,Servicio Local de Educación(SLE),Rural,15,151,15101,Arica,115,...,4,12,26,73,38.211631,53.991890,54.052684,76.130123,14.918749,69.341630
21,33,7,ESCUELA CARLOS CONDELL DE LA HAZA,Servicio Local de Educación(SLE),Rural,15,151,15101,Arica,38,...,2,11,5,20,45.266907,53.447100,57.669750,73.994210,15.235351,68.991242
22,35,3,ESCUELA GRAL. MANUEL BAQUEDANO,Servicio Local de Educación(SLE),Rural,15,151,15101,Arica,31,...,3,4,10,14,30.990030,49.248715,59.938797,72.219552,14.406247,61.889415
24,38,8,ESCUELA ESPAÑA,Servicio Local de Educación(SLE),Rural,15,151,15101,Arica,81,...,5,12,23,41,39.817246,54.180971,52.156412,73.616285,14.893180,66.380821
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5988,40146,3,COLEGIO SAN RAMON,Particular Subvencionado,Rural,6,61,6101,Rancagua,16,...,3,0,6,7,23.140398,49.336639,41.511876,70.195586,5.483696,20.457593
6056,40303,2,ESCUELA ESPECIAL DE LENGUAJE CHOCOLATITO,Particular Subvencionado,Rural,4,43,4304,Punitaqui,9,...,0,3,0,6,26.951294,44.461114,36.588058,71.115458,48.720241,25.257943
6117,41129,9,COLEGIO TALHUEN,Particular Subvencionado,Rural,7,71,7105,Maule,23,...,3,4,4,12,24.332021,46.343884,41.227328,70.847156,17.971210,5.401138
6131,41619,3,ESCUELA DE LENGUAJE PALABRITAS,Particular Subvencionado,Rural,7,74,7405,Retiro,18,...,1,1,2,14,26.089497,47.489950,51.713229,72.913270,50.872443,8.419314


In [114]:
dc_p_r = df_p_r_c[["ID_COMUNA_ESTABLE","cantidad establecimientos rurales"]].copy()
dc_p_u = df_p_u_c[["ID_COMUNA_ESTABLE","cantidad establecimientos urbanos"]].copy()
dc_pc_r = df_pc_r_c[["ID_COMUNA_ESTABLE","cantidad establecimientos rurales"]].copy()
dc_pc_u = df_pc_u_c[["ID_COMUNA_ESTABLE","cantidad establecimientos urbanos"]].copy()
dc_sc_r = df_sc_r_c[["ID_COMUNA_ESTABLE","cantidad establecimientos rurales"]].copy()
dc_sc_u = df_sc_u_c[["ID_COMUNA_ESTABLE","cantidad establecimientos urbanos"]].copy()
dc_m_r = df_m_r_c[["ID_COMUNA_ESTABLE","cantidad establecimientos rurales"]].copy()
dc_m_u = df_m_u_c[["ID_COMUNA_ESTABLE","cantidad establecimientos urbanos"]].copy()
dc_m_u

,ID_COMUNA_ESTABLE,cantidad establecimientos urbanos
0,15101,2449
1,15101,2449
2,15101,2449
3,15101,2449
4,15101,2449
...,...,...
2613,11101,2449
2614,13503,2449
2615,13201,2449
2616,9109,2449


In [124]:
rurales = (pd.concat([dc_p_r, dc_pc_r, dc_sc_r, dc_m_r]).groupby("ID_COMUNA_ESTABLE", as_index=False).sum())

In [126]:
urbanos = (pd.concat([dc_p_u, dc_pc_u, dc_sc_u, dc_m_u]).groupby("ID_COMUNA_ESTABLE", as_index=False).sum())

In [129]:
estab_comunal = rurales.merge(urbanos, on="ID_COMUNA_ESTABLE", how="outer")
estab_comunal.drop_duplicates(subset="ID_COMUNA_ESTABLE")


,ID_COMUNA_ESTABLE,cantidad establecimientos rurales,cantidad establecimientos urbanos
0,1101,17287.0,727287.0
1,1107,15964.0,407450.0
2,1401,53217.0,NaN
3,1402,29507.0,NaN
4,1403,14737.0,NaN
...,...,...,...
340,16301,88089.0,208363.0
341,16302,84574.0,75801.0
342,16303,44325.0,24681.0
343,16304,20506.0,15650.0


In [130]:
estab_comunal.isna().sum()

ID_COMUNA_ESTABLE                     0
cantidad establecimientos rurales    46
cantidad establecimientos urbanos    31
dtype: int64

### 5.3 Unión de bases de datos y polígonos (0.4 pts)

Finalmente, vamos a generar el DataFrame `agg_comunal` con la información de los dos previos DataFrames creados.
Combine la información de las dos secciones anteriores para crear su base de datos unificada. Por último, agregue la columna `geometry` con la información geográfica de cada comuna.

In [132]:
# respuestas
comunas_chile = gp.read_file("data/Comunas.zip")
comunas_chile.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 346 entries, 0 to 345
Data columns (total 12 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   objectid    346 non-null    int64   
 1   shape_leng  346 non-null    float64 
 2   dis_elec    346 non-null    int32   
 3   cir_sena    346 non-null    int32   
 4   cod_comuna  346 non-null    int32   
 5   codregion   346 non-null    int32   
 6   st_area_sh  346 non-null    float64 
 7   st_length_  346 non-null    float64 
 8   Region      346 non-null    object  
 9   Comuna      346 non-null    object  
 10  Provincia   346 non-null    object  
 11  geometry    346 non-null    geometry
dtypes: float64(3), geometry(1), int32(4), int64(1), object(3)
memory usage: 27.2+ KB


In [ ]:
agg_comunal = estab_comunal.merge(ivm_comunal,on="ID_COMUNA_ESTABLE",how="outer")
agg_comunal["geometry"] = comunas_chile["geometry"]
agg_comunal


,ID_COMUNA_ESTABLE,cantidad establecimientos rurales,cantidad establecimientos urbanos,ID_REGION_ESTABLE,DS_COMUNA_ESTABLE,IVM Promedio Parvulo,IVM Promedio Primer Ciclo,IVM Promedio Segundo Ciclo,IVM Promedio Media,geometry
0,1101,17287.0,727287.0,1,Iquique,55.460412,43.522262,38.432973,20.252681,"POLYGON ((-7992818.789 -4056669.319, -7992731...."
1,1107,15964.0,407450.0,1,Alto Hospicio,55.460412,43.522262,38.432973,20.252681,"POLYGON ((-7831651.813 -4022443.109, -7831596...."
2,1401,53217.0,NaN,1,Pozo Almonte,55.460412,43.522262,38.432973,20.252681,"POLYGON ((-7892616.357 -4060673.364, -7892621...."
3,1402,29507.0,NaN,1,Camiña,55.460412,43.522262,38.432973,20.252681,"POLYGON ((-7906457.717 -4051723.734, -7906266...."
4,1403,14737.0,NaN,1,Colchane,55.460412,43.522262,38.432973,20.252681,"POLYGON ((-8121756.186 -4645860.295, -8121644...."
...,...,...,...,...,...,...,...,...,...,...
340,16301,88089.0,208363.0,16,San Carlos,55.460412,43.522262,38.432973,20.252681,"POLYGON ((-8054569.835 -4420582.244, -8054531...."
341,16302,84574.0,75801.0,16,Coihueco,55.460412,43.522262,38.432973,20.252681,"POLYGON ((-8041950.854 -4349201.562, -8041735...."
342,16303,44325.0,24681.0,16,Ñiquén,55.460412,43.522262,38.432973,20.252681,"POLYGON ((-7931328.549 -4363544.179, -7931320...."
343,16304,20506.0,15650.0,16,San Fabián,55.460412,43.522262,38.432973,20.252681,"POLYGON ((-8026032.186 -4319505.012, -8025933...."


## 6. Análisis de vulnerabilidad por comuna (1.5 puntos)

### 6.1 Mapa de calor (0.8 pts)

Genere mapas de calor de las comunas de Chile según el IVM promedio de los establecimientos educacionales. Realice esto para cada etapa de enseñanza (Parvularia, Básica 1, Básica 2 y Media). Analice y comente sus resultados:
* ¿Cómo se diferencia la vulnerabilidad en distintos sectores del país según nivel de enseñanza?

In [ ]:
# Respuesta
comunas_chile = gp.read_file("data/Comunas.zip")
comunas_chile

,objectid,shape_leng,dis_elec,cir_sena,cod_comuna,codregion,st_area_sh,st_length_,Region,Comuna,Provincia,geometry
0,48,170038.624165,16,8,6204,6,9.685774e+08,206184.271675,Región del Libertador Bernardo O'Higgins,Marchigüe,Cardenal Caro,"POLYGON ((-7992818.789 -4056669.319, -7992731...."
1,29,125730.104795,15,8,6102,6,4.157446e+08,151911.576827,Región del Libertador Bernardo O'Higgins,Codegua,Cachapoal,"POLYGON ((-7831651.813 -4022443.109, -7831596...."
2,30,63026.084422,15,8,6103,6,1.448565e+08,76355.326122,Región del Libertador Bernardo O'Higgins,Coinco,Cachapoal,"POLYGON ((-7892616.357 -4060673.364, -7892621...."
3,31,89840.903562,15,8,6104,6,3.256572e+08,108874.623150,Región del Libertador Bernardo O'Higgins,Coltauco,Cachapoal,"POLYGON ((-7906457.717 -4051723.734, -7906266...."
4,78,122626.493264,23,11,9121,9,6.990727e+08,156680.410681,Región de La Araucanía,Cholchol,Cautín,"POLYGON ((-8121756.186 -4645860.295, -8121644...."
...,...,...,...,...,...,...,...,...,...,...,...,...
341,255,130240.555805,19,10,16305,16,8.751006e+08,160459.137929,Región de Ñuble,San Nicolás,Punilla,"POLYGON ((-8041950.854 -4349201.562, -8041735...."
342,253,311061.932415,19,10,16304,16,2.393007e+09,388109.613857,Región de Ñuble,San Fabián,Punilla,"POLYGON ((-7931328.549 -4363544.179, -7931320...."
343,245,159257.906431,19,10,16303,16,7.638173e+08,195593.933727,Región de Ñuble,Ñiquén,Punilla,"POLYGON ((-8026032.186 -4319505.012, -8025933...."
344,243,157467.049667,19,10,16104,16,1.044405e+09,196962.664313,Región de Ñuble,El Carmen,Ñuble,"POLYGON ((-7986315.104 -4415272.935, -7986309...."


In [ ]:
gdf = comunas_chile.merge(, on="ID_COMUNA_ESTABLE", how="left")


KeyError: 'ID_COMUNA_ESTABLE'

Comentarios y análisis:

### 6.2 Relación entre ruralidad comunal y vulnerabilidad (0.7 pts)

En la sección **5.2** definimos el **porcentaje de ruralidad** de cada comuna según la cantidad de establecimientos rurales en cada una. Analice y visualice los datos en `agg_comunal` para responder la siguiente pregunta:

* ¿Hay alguna correlación entre el IVM promedio y el porcentaje de ruralidad en las comunas de Chile?

In [ ]:
# respuestas

## 7 Pregunta Bono (2 punto)

* Proponga una pregunta de análisis que le interese responder con los datos de cualquiera de las bases de datos leídas o generadas durante esta Tarea.
Respóndala de la misma forma que las preguntas anteriores, generando una o varias visualizaciones y analizando sus resultados. Justifique 

**Recomendación**: Trate de proponer una pregunta que justifique el uso de agregación de datos y visualización. En otras palabras, que
no pueda ser respondida directamente sin el uso de gráficos (por ejemplo, *"¿cuál es el establecimiento con mayor IVM?"* puede ser respondida con una búsqueda rápida en el DataFrame).

**Pregunta propuesta:**

In [ ]:
# respuesta